stable_baselines3/common/callbacks.py

```diff
@@ -435,3 +435,47 @@ def _on_step(self) -> bool:
             self.last_time_trigger = self.num_timesteps
             return self._on_event()
         return True
+
+
+class StopTrainingOnMaxEpisodes(BaseCallback):
+    """
+    Stop the training once a maximum number of episodes are played.
+
+    For multiple environments presumes that, the desired behavior is that the agent trains on each env for ``max_episodes``
+    and in total for ``max_episodes * n_envs`` episodes.
+
+    :param max_episodes: (int) Maximum number of episodes to stop training.
+    :param verbose: (int)
+    """
+
+    def __init__(self, max_episodes: int, verbose: int = 0):
+        super(StopTrainingOnMaxEpisodes, self).__init__(verbose=verbose)
+        self.max_episodes = max_episodes
+        self._max_episodes_per_env = max_episodes
+        self.episodes_played = 0
+        self._first_step = True
+
+    def _on_step(self) -> bool:
+        # Checking for both 'done' and 'dones' keywords because:
+        # Some models use keyword 'done' (e.g.,: SAC, TD3, DQN, DDPG)
+        # While some models use keyword 'dones' (e.g.,: A2C, PPO)
+        done_array = np.array(self.locals.get("done") if self.locals.get("done") is not None else self.locals.get("dones"))
+        self.episodes_played += np.sum(done_array).item()
+
+        # Sets max episodes per env in first step to avoid doing this operation multiple times
+        if self._first_step:
+            self._max_episodes_per_env = self.max_episodes * done_array.size
+            self._first_step = False
+
+        continue_training = bool(self.episodes_played < self._max_episodes_per_env)
+
+        if self.verbose > 0 and not continue_training:
+            avg_eps_per_env = self.episodes_played // done_array.size
```

Comment:
why not a float division? (and then use `"{:.2f}"` for formatting)

Reply:
My rationale was that the user would expect the training to run for `max_episodes` for each env, and that a few episodes more would not be a problem. Hence, I find that it may be confusing for the user to expect an exact number of episodes during training, and end up with a float number of let's say 10.1 episodes. Also, I believe an exact division will be more common than one that results in a real value, and there is not much information gain in knowing the exact average. What do you think?

Reply:
ok, we just need to make sure that `self.episodes_played // done_array.size` would be greater than one, otherwise the message will be misleading

Reply:
As I see, this could only happen if the argument `max_episodes` passed by the user is lower than `num_envs`. So, do you prefer that I implicitly set `max_episodes == num_envs` in these cases, and log a warning? Or throw an exception if `max_episodes < num_envs`? Or do you have any other suggestions?

Reply:
I thought it should run at least `max_episodes * num_envs`, no? (at least that what is written)

Reply:
Yes, that is right and the division `n_episodes // num_envs` occurs only if n_episodes >= max_episodes * num_envs. Also, this division will only be less than 1 if `max_episodes < num_envs`.

I am not sure I understand the issue here, so I can just do as you first suggested and change the division to a float one, no problem.